  Fix inner class test so that it no longer fails when run from IDEA.  Summary: The inner class TimeoutTest$ThreadGuardedTest used to fail when run from IntelliJ. The problem is that the TimeoutTest$ThreadGuardedTest test is actually test *data*, not test code.  (It's a sample test to verify how Buck's test runner works.) Unfortunately, if this was test *data*, it would be nasty to build and debug. This is a simple hack to only allow the tests to run as part of the test that uses it as data.  Test Plan: Before this change, run all the tests in com.facebook.buck.testrunner from IntelliJ by right-clicking on the package in the project sidebar and choosing "Run Tests in 'com.facebook.buck.testrunner'".  Observe that the tests in the TimeoutTest$ThreadGuardedTest inner class are picked up by the runner and that they fail (by timing out).  After the change, observe that the test is ignored (immediate assumption violation), but the outer class (TimeoutTest) that uses it still works as expected.  Reviewed By: yiding  fbshipit-source-id: d7b28b5  